cli/src/command/wallet.rs

```diff
@@ -111,10 +113,30 @@ pub enum WalletCommand {
     Sync,
 }
 
+impl WalletCommand {
+    pub fn is_unlock_command(&self) -> bool {
+        match self {
+            Self::Accounts
+            | Self::Backup { .. }
+            | Self::ChangePassword { .. }
+            | Self::NewAccount { .. }
+            | Self::NodeInfo
+            | Self::SetNodeUrl { .. }
+            | Self::SetPow { .. }
+            | Self::Sync => true,
+            Self::Init(_)
+            | Self::MigrateStrongholdSnapshotV2ToV3 { .. }
+            | Self::Mnemonic { .. }
+            | Self::Restore { .. } => false,
+        }
+    }
+}
+
 #[derive(Debug, Clone, Args)]
 pub struct InitParameters {
+    // TODO: remove this field to make `InitParameters` independ from the secret manager being used?
```

Comment:
TODO

Reply:
Removed the TODO because achieving this (if desirable) requires more than just removing this field but a whole refactor of this struct.

Reply:
If you think it's worth doing, can we create an issue?

Reply:
not worth doing